src/include/cell_inline.h

```diff
@@ -970,6 +970,11 @@ __cell_unpack_window_cleanup(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk
      * No delete              txnid=MAX, ts=MAX,            txnid=MAX, ts=MAX,
      *                        durable_ts=NONE               durable_ts=NONE
      */
+    /* Don't reset the transaction ids in rollback to stable when called from recovery. */
+    if (F_ISSET(S2C(session), WT_CONN_RECOVERING) &&
+      F_ISSET(session, WT_SESSION_ROLLBACK_TO_STABLE))
```

Comment:
I feel we don't need to check for the `rts` session as it is possible that HS may get overwritten with new updates that are inserted into the history store during log reply on other tables.

Reply:
If we don't check for the RTS session flag then other tests fail with visibility issues because in recovery those tests transaction ids will not be reset.